apps/calendar/js/views/single_month.js

```diff
@@ -45,13 +46,16 @@ SingleMonth.prototype = {
 
   _renderDayHeaders: function() {
     var days = [];
-    var i = startsOnMonday ? 0 : -1;
-    while (++i < 7) {
+    var i;
+    var range = 7 - startDay;
+    for (i = startDay; i<7; i++) {
       days.push(this._dayHeader(i));
     }
 
-    if (startsOnMonday) {
-      days.push(this._dayHeader(0));
+    if (range < 7) {
```

Comment:
`range` is a confusing name.. maybe do just `if (startDay > 0)` specially since `range` is only used here..